file-upload-example/README.md

```diff
@@ -0,0 +1,139 @@
+# Appwrite + Angular = ❤️
+
+This example is to showcase [Appwrite's JS API](https://github.com/appwrite/sdk-for-js) with [Angular](https://angular.io/) by creating a file uploading application where you can upload files, see recently uploaded files, preview a file, download or delete files.
+
+Please note that in this tutorial I'll only cover Appwrite specific parts.
+
+## Sneak peek
+
+This is what we will go through in details:
+
+![sneak peak](end-result.png)
+
+## Prerequisites
+
+-   A Recent Version of NodeJS
+-   Npm or yarn (I'll you npm for this tutorial for the sake of simplicity)
+-   [A locally running appwrite instance](https://appwrite.io/docs/installation).
+-   Create a user for testing
+
+## Development server
+
+Run `npm start` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
+
+> You can also change the port if `4200` is already in use with `npm start -- --port=4300`
+
+## Getting Started
+
+``` shell
+git clone git@github.com:appwrite/demos-for-angular.git
+cd file-upload-example
+npm install
+npm start
+```
+
+Open the folder with your favorite code editor.
+
+## Setting project and backend url
+
+Let's look at our `src/environments/environment.ts`, this is where Angular stores all of our environment variables, so here is where we set `project` and `endpoint` for Appwrite
+
+Here is an example that I used:
+
+```ts
+export const environment = {
+  production: false,
+  endpoint: 'http://localhost/v1', // replace with your own appwrite server ip
+  project: '5f7860217c7e7' // replace with your own appwrite project id
+};
+```
+
+## Working with Appwrite
+
+For working with Appwrite we will use a a service (`src/app/appwrite.service.ts`). Most of the service's methods is just a wrapper around Appwrite's SDK Storage API methods.   
+I'll wrap every promise around an `Observable` to be consistent with Angular's built-in `HttpClient`.
+
+### Detail about our service
+
+#### Initializing the SDK
+
+```ts
+  // ...
+  constructor() {
+    this.sdk.setEndpoint(environment.endpoint);
+    this.sdk.setProject(environment.project);
+  }
+  // ...
+```
+
+With this as soon as the service is creating the `endpoint` and `project` is also set and since our service is provided in root it will be only created once across the whole app. 
```

Comment:
```suggestion
With the above code, as soon as the service is created, the `endpoint` and `project` is also set. Since our service is provided in root it will be only created once across the whole app. 
```